### PR TITLE
Labs23 T1 Search Bar

### PR DESCRIPTION
**Pull request by SilverMaiden**
Implemented search bar functionality, allowing users to search through projects by name.

## Type of change

Created a FilterBar component, along with a new filtering query. The results of this new filtering query are now set to the official state in "ProjectList/index.jsx". Context was implemented via ProjectSearchContext.

- [x] New feature (non-breaking change which adds functionality)

## Change Status

- [x] Complete, tested, ready to review and merge

# How Has This Been Tested?

- [x] Tested by multiple peers

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My code has been reviewed by at least one peer
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] There are no merge conflicts

### CODE REVIEW

**Review by nomadkitty — APPROVED** [review]
Great job

**Review by ksemenza — APPROVED** [review]

File: src/components/App.js [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import React, { useEffect } from 'react';
 import { useLocation, Route } from 'react-router-dom';
 import { SecureRoute, ImplicitCallback } from '@okta/okta-react';
+import { ProjectSearchProvider} from '../contexts/labs23-t1-filterbar-context';
+
```
Comment (ryan-hamblin) [inline src/components/App.js:5]: I don't think this extra space is necessary.

File: src/components/App.js [hunk 2]
```diff
@@ -18,8 +20,10 @@ const App = () => {
   return [
     <LabelProvider>
       <ColumnProvider>
+      <ProjectSearchProvider>
         <Route path="/implicit/callback" component={ImplicitCallback} />
         <SecureRoute path="/" component={Authorization} />
+        </ProjectSearchProvider>
```
Comment (ryan-hamblin) [inline src/components/App.js:26]: There are some indentation problems here.

File: src/components/Authorization/index.jsx [hunk 3]
```diff
@@ -2,6 +2,8 @@ import React, { useState, useEffect, useCallback } from 'react';
 import { withAuth, SecureRoute } from '@okta/okta-react';
 import { Switch } from 'react-router-dom';
 
+import FilterBar from '../../components/FilterBar/FilterBar';
+
```
Comment (ryan-hamblin) [inline src/components/Authorization/index.jsx:6]: Let's remove this extra line break

File: src/components/FilterBar/FilterBar.js [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+import { useQuery } from "urql";
+import gql from "graphql-tag";
+
+import Project from "./Project";
+import './FilterBar.scss';
+
+
+const FilterBar = () => {
+  // const classes = useStyles();
```
Comment (ryan-hamblin) [inline src/components/FilterBar/FilterBar.js:14]: Please remove commented out code.

File: src/components/FilterBar/FilterBar.js [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+  const [searchFilter, setSearchFilter] = useState('');
+  const [filterList, setFilterList] = useState([]);
+
```
Comment (ryan-hamblin) [inline src/components/FilterBar/FilterBar.js:17]: Lot's of line breaks here.

File: src/components/FilterBar/FilterBar.js [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+  const handleSubmit = e => {
+    //hitting enter key submits immediately
+    e && e.preventDefault();
+    //clear the submit timer and submit form
+    window.clearTimeout(timer);
+    if (searchFilter !== '') {
+      console.log('send new or update query to BE')
+      //Setting searchTerm in projectSearchContext to value of searchFilter
+      //so it's accessible to "ProjectList/index.jsx"
```
Comment (ryan-hamblin) [inline src/components/FilterBar/FilterBar.js:38]: GREAT code comments.

File: src/components/FilterBar/FilterBar.js [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+
+
+  }//end handleSubmit
+
+  const removeTag = (item) => {
+    console.log("Remove tag from backend query");
+    let newFilterList = filterList.filter(ele => {
+      return ele !== item;
+    })
+    // console.log('newList: ', newFilterList);
```
Comment (ryan-hamblin) [inline src/components/FilterBar/FilterBar.js:57]: Remove the console.logs :)

File: src/components/FilterBar/filterBarQueries.js [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+      name
+      active
+      updatedAt
+      notes(orderBy: updatedAt_DESC) {
+        id
+        updatedAt
+      }
+      status {
+        id
+        # category {
```
Comment (ryan-hamblin) [inline src/components/FilterBar/filterBarQueries.js:17]: Please remove commented out codes

File: src/components/ProjectList/index.jsx [hunk 9]
```diff
@@ -10,7 +12,20 @@ import Settings from '../Settings/Settings';
 // and returns a list of projects that they are authorized to view.
 
 const ProjectListView = () => {
-  const [state] = useQuery({ query });
+ 
+  const projectSearchContext = useContext(ProjectSearchContext);
+
+  // Currently using this test state to query data. Applies the filter no matter what,
+  // though initially the search term is an empty string, leading to all projects being shown.
```
Comment (ryan-hamblin) [inline src/components/ProjectList/index.jsx:19]: Love this!

File: src/components/ProjectList/index.jsx [hunk 10]
```diff
@@ -23,8 +38,11 @@ const ProjectListView = () => {
 
   if (fetching) {
     return <p>Loading...</p>;
+  } else {
+    console.log(data)
   }
 
+
   // const columns =
   //   data && data.programs[0].statuses && data.programs[0].statuses; // .filter(status => status.display);
```
Comment (ryan-hamblin) [inline src/components/ProjectList/index.jsx:47]: Remove me

File: src/components/ProjectList/missioncontrol-fe.code-workspace [hunk 11]
```diff
@@ -0,0 +1,7 @@
+{
+	"folders": [
+		{
+			"path": "/Users/SilverMaiden/programming/Lambda_School/mission-control-fe"
```
Comment (ryan-hamblin) [inline src/components/ProjectList/missioncontrol-fe.code-workspace:4]: formatting/spacing here is off.

**Review by ryan-hamblin — COMMENTED** [review]
This PR looks really good, there are some things to improve in terms of the code itself. I'd make sure that when you're requesting a review of your code, that the person who reviews actually dives deep into the files that you've written code in and does the review there. This provides for an active discussion around your hard work and pushes you to do a bit better as a developer. 

There are a few things you could do to fix a lot of this, 1.) Remove all the logs/commented out code and unnecessary spacing. 2.) Please make sure that the team installs a linter and code formatter to ensure that everyone's code gets formatted the same consistency throughout the app. 

Reviewers, PLEASE make sure that you do a thorough review of the code. Your job is not to praise, it's to critique/review the code itself. 

Overall GREAT stuff here. I'm so proud of this entire team for all the hard work you've put into your Labs experience.

**Pull request merged**